Bottom: 4b48d0ce43be23955352db3c5c731937dc81fc15
Top:    2c4d6f97cba6bcf49ff25b8d1f889164d8fd5c81
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-08-14 15:33:30 -0500

Report build status to Gitlab


---
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index 3def5922..bd8670cd 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -12,6 +12,7 @@ buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/cre
 buildsystem_ssh_root_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/root"                , length=20) }}'
 gitlab_buildbot_auth_id:             '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_client_id") }}'
 gitlab_buildbot_auth_secret:         '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_secret") }}'
+gitlab_buildbot_token:               '{{ lookup("password", playbook_dir + "/credentials/git/auth/buildbot_token") }}'
 gitlab_db_password:                  '{{ lookup("password", playbook_dir + "/credentials/git/db/user"                         , length=20) }}'
 gitlab_email_password:               '{{ lookup("password", playbook_dir + "/credentials/git/email/gitlab@novatech-llc.com"   , length=20) }}'
 gitlab_root_password:                '{{ lookup("password", playbook_dir + "/credentials/git/users/root"                      , length=20) }}'
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index d9b8f54b..72733708 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -20,7 +20,8 @@ RUN pip --no-cache-dir install \
     'buildbot-waterfall-view' \
     'buildbot-www' \
     'ldap3' \
-    'pyOpenSSL'
+    'pyOpenSSL' \
+    'txrequests'
 
 # Create buildbot user
 ARG BUILDBOT_UID=1000
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 1ac46a1a..32b0b5c0 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -60,6 +60,15 @@ mn = reporters.MailNotifier(
     )
 c['services'].append(mn)
 
+gl = reporters.GitLabStatusPush(
+    os.getenv('GITLAB_TOKEN'),
+    context="Buildbot CI",
+    baseURL=os.getenv("GITLAB_URI"),
+    verbose=True,
+    debug=True
+)
+c['services'].append(gl)
+
 ####### WEB SERVER
 userInfoProvider = util.LdapUserInfo(
         uri=os.getenv('LDAP_URI'),
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
index ef79a5b9..d1aa07d5 100644
--- a/roles/buildbot-master/defaults/main.yml
+++ b/roles/buildbot-master/defaults/main.yml
@@ -49,8 +49,9 @@ container_env:
   BUILDBOT_WORKER_PTXDIST: 'worker-ptxdist'
   BUILDBOT_WORKER_PTXDIST_PASS: '{{ buildbot_worker_ptxdist_password }}'
   GITLAB_URI: 'https://{{ gitlab_hostname }}/'
-  GITLAB_CLIENT_ID: '{{ gitlab_buildbot_auth_id }}'
-  GITLAB_CLIENT_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_OAUTH_ID: '{{ gitlab_buildbot_auth_id }}'
+  GITLAB_OAUTH_SECRET: '{{ gitlab_buildbot_auth_secret }}'
+  GITLAB_TOKEN: '{{ gitlab_buildbot_token }}'
   GITLAB_WEBHOOK_SECRET: '{{ buildbot_webhook_secret }}'
   LDAP_URI: 'ldap://{{ openldap_hostname }}:389/'
   LDAP_BIND_USER: 'cn=proxyagent,dc=novatech'
